pkg/kubelet/server/server_test.go

```diff
@@ -1506,3 +1515,33 @@ func TestTrimURLPath(t *testing.T) {
 		assert.Equal(t, test.expected, getURLRootPath(test.path), fmt.Sprintf("path is: %s", test.path))
 	}
 }
+
+func TestCadvisorMetrics(t *testing.T) {
+	mi := &cadvisorapi.MachineInfo{}
+	b, err := ioutil.ReadFile("testdata/machine-info.json")
+	require.NoError(t, err)
+	require.NoError(t, json.Unmarshal(b, mi))
+
+	containerInfoMap := map[string]*cadvisorapi.ContainerInfo{}
+	b, err = ioutil.ReadFile("testdata/requested-container-info.json")
+	require.NoError(t, err)
+	require.NoError(t, json.Unmarshal(b, &containerInfoMap))
+
+	fw := newServerTest()
+	fw.fakeKubelet.machineInfoFunc = func() (*cadvisorapi.MachineInfo, error) { return mi, nil }
+	fw.fakeKubelet.reqContainersInfoFunc = func(containerName string, options cadvisorapiv2.RequestOptions) (map[string]*cadvisorapi.ContainerInfo, error) {
+		return containerInfoMap, nil
+	}
+	defer fw.testHTTPServer.Close()
+
+	expected, err := os.Open("testdata/expected-cadvisor-metrics.txt")
+	require.NoError(t, err)
+	err = testutil.ScrapeAndCompare(fw.testHTTPServer.URL+"/metrics/cadvisor", expected)
+	require.NoError(t, err)
+
+	// Second call should lead the same result.
+	expected, err = os.Open("testdata/expected-cadvisor-metrics.txt")
+	require.NoError(t, err)
+	err = testutil.ScrapeAndCompare(fw.testHTTPServer.URL+"/metrics/cadvisor", expected)
```

Comment:
why do we need to scrape and compare twice? In other words, why would it be different?

Reply:
I guess this was mostly added as I safeguard. The only reason I could see why it would defer is if there was a problem generating the metrics that are generated at scrape time (ConstMetrics in Prometheus library). But since the test data doesn't change, if there was a problem it would have been detected by the first scrape.

I am fine removing it.